src/Devices/Ethernet_TC.cpp

```diff
@@ -0,0 +1,32 @@
+#include "Ethernet_TC.h"
+using namespace std;
+
+TC_ethernet *TC_ethernet::_instance = nullptr;
+
+TC_ethernet::TC_ethernet() {
+  pinMode(4, OUTPUT);
+  digitalWrite(4, HIGH);
+  if (Ethernet.begin(mac) == 0) {
+    Serial.println(F("Failed to configure Ethernet using DHCP"));
+    Ethernet.begin(mac, defaultIP);
+    IP = defaultIP;
+  }
+
+  defaultIP = IPAddress(192, 168, 1, 2);
+  time_serverIP = IPAddress(132, 163, 97, 1);
+}
+
+TC_ethernet *TC_ethernet::getInstance() {
+  if (_instance == nullptr) {
+    _instance = new TC_ethernet;
+  }
+  return _instance;
+}
+
+void TC_ethernet::renewDHCPLease() {
+  // unsigned long current_millis = millis();
```

Comment:
Either take out the commented-out code or (even better), keep track of the previous lease time and the LEASE_INTERVAL as a private instance variable and constant so that this method can be called frequently and it will decide when it is time to renew (rather than that knowledge being in the caller).